docs/reference/analysis/tokenfilters/minhash-tokenfilter.asciidoc

```diff
@@ -20,3 +54,80 @@ The following are settings that can be set for a `min_hash` token filter.
 bucket to its circular right. Only takes effect if hash_set_size is equal to one.
 Defaults to `true` if bucket_count is greater than one, else `false`.
 |=======================================================================
+
+Some points to consider while setting up a `min_hash` filter:
+
+* `min_hash` filter input tokens should typically be k-words shingles produced
+from <<analysis-shingle-tokenfilter,shingle token filter>>.  You should
+choose `k` large enough so that the probability of any given shingle
+occurring in a  document is low. At the same time, as
+internally each shingle is hashed into to 128-bit hash, you should choose
+`k` small enough so that all possible
+different k-words shingles can be hashed to 128-bit hash with
+minimal collision. 5-word shingles typically work well.
```

Comment:
@cbuescher  I took an advice on `5-word shingle` from the MinHash filter [sourcecode](https://github.com/apache/lucene-solr/blob/master/lucene/analysis/common/src/java/org/apache/lucene/analysis/minhash/MinHashFilter.java#L40) in Lucene

Reply:
Thats interesting, would you mind linking to that source?

Reply:
@cbuescher Thanks for the suggetion. I opted not to include  the link to this source, as I am afraid as the sourcecode changes this link becomes invalid.

Reply:
In the original PR that adds `min_hash`, it looks like we were not sure about the 5 word suggestion, and instead encouraged 2 word shingles: https://github.com/elastic/elasticsearch/pull/20206#issuecomment-243129857. It would be nice if there was a reference or set of experiments to help confirm a good default value... I didn't manage to find one in a quick search, but will keep a lookout. The right choice seems like it would depend on the use case as well (for example similarity search vs. duplicate detection).

Reply:
@jtibshirani  Thanks a lot for the review.  I think the best for now is to remove this line completely "5-word shingles typically work well.", as there are conflicting suggestions what shingle size works best.  Once we have better sources (external or from our own experiments), we can add shingle size suggestions to the file. Is this fine with you?

Reply:
This sounds like a good plan to me!